numpy/core/tests/test_multiarray.py

```diff
@@ -3299,10 +3299,8 @@ def test__complex__(self):
             msg = 'dtype: {0}'.format(dt)
             ap = complex(a)
             assert_equal(ap, a, msg)
-            bp = complex(b)
-            assert_equal(bp, b, msg)
-            cp = complex(c)
-            assert_equal(cp, c, msg)
+            assert_warns(DeprecationWarning, complex, b)
+            assert_warns(DeprecationWarning, complex, c)
```

Comment:
You should be able to simultaneously assert that this warns and check its value here